### PR TITLE
Bracelet management contract (bag of seeds) #197

### PR DESCRIPTION
**Pull request by IanMendozaJaimes**
- Added new contract bracelet
- Added tests

The `transfer` action, enforces the usage of the `bracelet` permission. The `bracelet` permission should be created by the user as well as the linkAuth. There is an example of this in the tests.

I didn't create the contract in helper.js for testnet nor mainnet, just in case

### CODE REVIEW

File: src/seeds.bracelet.cpp [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  _transfer(to, quantity, memo);
+}
+
+void bracelet::init_balance (name account) {
+  auto bitr = balances.find(account.value);
+  if (bitr == balances.end()) {
+    balances.emplace(_self, [&](auto & item){
+      item.account = account;
+      item.balance = asset(0, utils::seeds_symbol);
+      item.is_freezed = false;
```
Comment (juliolrmonteiro) [inline src/seeds.bracelet.cpp:78]: I would rename this to "is_frozen"

File: include/seeds.bracelet.hpp [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    void check_freeze(name account);
+
+    DEFINE_USER_TABLE
+
+    DEFINE_USER_TABLE_MULTI_INDEX
+
+    TABLE balance_table {
+      name account;
+      asset balance;
+      bool is_freezed;
```
Comment (juliolrmonteiro) [inline include/seeds.bracelet.hpp:56]: Better rename to "is_frozen" :)

File: src/seeds.bracelet.cpp [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    item.balance -= quantity;
+  });
+
+  balances.modify(bcitr, _self, [&](auto & item){
+    item.balance -= quantity;
+  });
+}
+
+void bracelet::_deposit (asset quantity) {
+  check_asset(quantity);
```
Comment (juliolrmonteiro) [inline src/seeds.bracelet.cpp:121]: Should this be `utils::check_asset`?

File: src/seeds.bracelet.cpp [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+  token::transfer_action action{contracts::token, {_self, "active"_n}};
+  action.send(_self, contracts::bank, quantity, "");
```
Comment (juliolrmonteiro) [inline src/seeds.bracelet.cpp:124]: Looks like the memo is empty here on this transfer, should it mention it's a "bracelet deposit"?

File: src/seeds.bracelet.cpp [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+void bracelet::_transfer (name beneficiary, asset quantity, string memo) {
+  utils::check_asset(quantity);
+  token::transfer_action action{contracts::token, {contracts::bank, "active"_n}};
+  action.send(contracts::bank, beneficiary, quantity, memo);
+}
+
+void bracelet::check_freeze (name account) {
+  auto bitr = balances.get(account.value, "bracelet: no balance object found");
+  check(!bitr.is_freezed, "bracelet: account is freezed");
```
Comment (juliolrmonteiro) [inline src/seeds.bracelet.cpp:140]: Just rename all instances of "freezed" to "frozen" ;)

**Review by juliolrmonteiro — CHANGES_REQUESTED** [review]
This is very simple and well written, just small checks and requesting var names review.

File: src/seeds.bracelet.cpp (hunk 5 above)
Comment (IanMendozaJaimes) [reply src/seeds.bracelet.cpp:140]: Yes, you're right, is_frozen is a better name

File: src/seeds.bracelet.cpp (hunk 4 above)
Comment (IanMendozaJaimes) [reply src/seeds.bracelet.cpp:124]: ok, I will change it. What about the memo in the `withdraw` action?, I left it empty as well for the moment

File: src/seeds.bracelet.cpp (hunk 3 above)
Comment (IanMendozaJaimes) [reply src/seeds.bracelet.cpp:121]: yes, I forgot the namespace, I'll change it

**Comment by IanMendozaJaimes** [comment]
I made the requested changes.

File: src/seeds.bracelet.cpp (hunk 4 above)
Comment (juliolrmonteiro) [reply src/seeds.bracelet.cpp]: Perfect, that's good for now. :)

**Review by juliolrmonteiro — APPROVED** [review]
Looks good to me. Thanks!

**Review by n13 — APPROVED** [review]

**Pull request merged**